flask_multipass_cern.py

```diff
@@ -170,20 +197,53 @@ def _fix_phone(self, data):
             return
         data['telephone1'] = self.settings['phone_prefix'] + phone
 
-    def _extract_extra_data(self, data):
-        return {'cern_person_id': data.pop('cernPersonId', None)}
+    def _extract_extra_data(self, data, default=None):
+        return {'cern_person_id': data.pop('cernPersonId', default)}
 
     def get_identity_from_auth(self, auth_info):
-        data = self._fetch_identity_data(auth_info)
+        upn = auth_info.data.get('cern_upn')
```

Comment:
IIRC `sub` contains it as well and is the more standard field. (and I'm using `sub` for the mattermost login and never had problems with it)